### ISSUE TITLE
[BUG] - 5e - Currencies are broken

### ISSUE THREAD

**Issue by Jabos90** [fantasycalendar/FoundryVTT-ItemPiles#602]
**Are you on the latest version of Item Piles?**
Yes

**Have you tried to reset Item Piles' system settings?**
No (But I did try this on a new world, twice, I presume this resets settings?)

**Describe the bug**
Any UI that displays a currency doesn't update _at all_, unless you re-open the window.

**To Reproduce**

(DM, client)
1. Click on '+ Add Currency' on Item Pile/Merchant
2. Set any currency to any value
3. Click on 'Submit'
4. Currency doesn't change until window is re-opened

(DM, client)
1. Click on '+ Add Currency' on Item Pile
2. Set any currency to any value
3. Click on 'Submit'
4. Click on 'Update'
5. Currency doesn't change, even if window is re-opened

(DM, client)
1. Open an Item Pile with currency
2. Click on 'Split currency among players'
3. Nothing happens

(Player, browser)
1. Open an Item Pile with currency
2. Click on 'Take' on currency
3. Currency is added to character, but not removed from the pile
**4. Player can keep clicking 'Take' however many times they want, getting more money every time**

(Player, browser)
1. Open a merchant
2. Sell/Buy anything
3. Currency doesn't change, until the window is re-opened (purely visual, currency still removed from character)

**Expected behavior**
1. The currency should increase by the values set in '+ Add Currency' dialog.
2. The currency should increase by the values set in '+ Add Currency' dialog.
3. The currency should be split evenly amongst players, and the distributed currency should be removed from the pile.
4. The currency should be removed from the pile immediately, and player should not be able to pick up the same currency repeatedly
5. The UI should reflect the money carried by the character

**Setup:**
- Foundry Version: 11.315
- System Version: DND5e 3.2.1
- Item Piles Version: 3.0.4
- Browser & Version: Firefox, 127.0.2

**Active modules:**
- Item Piles
- libwrapper
- socketli

**Additional context**
Not sure if it's related, or even a bug at all, but chat output seems inconsistent. Buying items from a merchant outputs into chat, but selling items does not. During our session, currency being taken only generated a chat card once (despite all players taking money multiple times).

**Comment by Jabos90** [comment]
So I tested and prepped this last night, but didn't have time to post. It seems there was a patch during that break that fixed the most of these issues, namely 1, 2, 4, and 5.

However, problem 3 (DM splits currency) still remains, although I didn't have time to check in a new world just yet.
I also noticed that if you click "split currency" a couple of times, and then show the pile to the player, they will still see the money, but it says "no share left".

I also tried adding 5 gold to a pile, then showed it to a player, and they only saw "1/2". They could take both coins, but nothing more. For the DM, 3 gold still remained.

**Comment by Haxxer** [comment]
When you do have the time, re-test 3 again - I believe it was fixed in the latest update

**Comment by Jabos90** [comment]
It appears you're right. I tested it again, and while it didn't work at first, I did notice that the "Split currency" button was _every so slightly_ greyed out. So I went searching in the settings and found the "Split only with active players" option, which seemed to have enabled it. So I guess it was just an issue of my actual players not being online, my bad.

I also had an issue with players opening merchants after a scene switch, but that also appears to have been fixed in the latest version. So false alarm, I guess, I'll close the issue, thanks.
